src/core/dbus-manager.c

```diff
@@ -1409,16 +1409,39 @@ static int dump_impl(
 
         _cleanup_free_ char *dump = NULL;
         Manager *m = ASSERT_PTR(userdata);
+        bool check_ratelimit = false;
         int r;
 
         assert(message);
 
-        /* Anyone can call this method */
-
+        /* 'status' access is the bare minimum always needed for this */
         r = mac_selinux_access_check(message, "status", error);
         if (r < 0)
                 return r;
 
+        /* Reload/reexec are also slow but non-destructive, use the same access check for dumps which, given
+         * the large amount of data to fetch, can stall for quite some time. If the caller is not privileged
+         * use rate limiting. */
+        r = mac_selinux_access_check(message, "reload", error);
+        if (r < 0)
+                check_ratelimit = true;
+
+        r = bus_verify_dump_async(m, message, error);
```

Comment:
if you came to the conclusion that ratelimiting is advised already above, this (very slow, because roundtrip involving) call should be skipped